Content.Client/Tabletop/TabletopSystem.cs

```diff
@@ -0,0 +1,303 @@
+            {
+                MinWidth = 500,
+                MinHeight = 436,
+                Title = msg.Title
+            };
+
+            _window.OnClose += OnWindowClose;
+        }
+
+        private void HandleComponentState(EntityUid uid, TabletopDraggableComponent component, ComponentHandleState args)
+        {
+            if (args.Current is not TabletopDraggableComponentState state) return;
+
+            component.DraggingPlayer = state.DraggingPlayer;
+        }
+
+        private void OnWindowClose()
+        {
+            StopDragging();
+            _window = null;
+        }
+
+        private bool OnUse(in PointerInputCmdHandler.PointerInputCmdArgs args)
+        {
+            return args.State switch
+            {
+                BoundKeyState.Down => OnMouseDown(args),
+                BoundKeyState.Up => OnMouseUp(args),
+                _ => false
+            };
+        }
+
+        private bool OnMouseDown(in PointerInputCmdHandler.PointerInputCmdArgs args)
+        {
+            // Return if no player entity
+            if (_playerManager.LocalPlayer is not { ControlledEntity: { } playerEntity }) return false;
+
+            // Return if can not see table or stunned/no hands
+            if (!CanSeeTable(playerEntity, _table) || StunnedOrNoHands(playerEntity))
+            {
+                return false;
+            }
+
+            // Set the entity being dragged and the viewport under the mouse
+            if (!EntityManager.TryGetEntity(args.EntityUid, out var draggedEntity))
+            {
+                return false;
+            }
+
+            // Make sure that entity can be dragged
+            if (!draggedEntity.HasComponent<TabletopDraggableComponent>())
+            {
+                return false;
+            }
+
+            // Try to get the viewport under the cursor, then start dragging the entity
+            if (_uiManger.MouseGetControl(args.ScreenCoordinates) as ScalingViewport is { } viewport)
+            {
+                StartDragging(draggedEntity, viewport);
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool OnMouseUp(in PointerInputCmdHandler.PointerInputCmdArgs args)
+        {
+            StopDragging();
+            return false;
+        }
+
+        #endregion
+
+        #region Utility
+
+        /**
+         * <summary>Whether the table exists, is in range and the player is alive.</summary>
+         * <param name="playerEntity">The player entity to check.</param>
+         * <param name="table">The table entity to check.</param>
+         */
+        private static bool CanSeeTable(IEntity playerEntity, IEntity? table)
+        {
+            if (table == null) return false;
+
+            var alive = playerEntity.TryGetComponent<MobStateComponent>(out var mob) && mob.IsAlive();
+            var inRange = playerEntity.InRangeUnobstructed(table);
+
+            return alive && inRange;
+        }
```

Comment:
why can't this use ActionBlocker?

Reply:
I'm 99% sure ActionBlocker can't be used here as I need different behaviour for when you get stunned/have no hands vs when you die. Same goes for the other comment.